testnet/stacks-node/src/config.rs

```diff
@@ -394,6 +398,28 @@ lazy_static! {
 }
 
 impl Config {
+    /// get the up-to-date burnchain from the config
+    pub fn get_burnchain_config(&self) -> Result<BurnchainConfig, String> {
+        if let Some(path) = &self.config_path {
+            let config_file = ConfigFile::from_path(path.as_str())?;
+            let config = Config::from_config_file(config_file)?;
+            Ok(config.burnchain)
+        } else {
+            Ok(self.burnchain.clone())
+        }
+    }
+
+    /// get the up-to-date miner options from the config
+    pub fn get_miner_config(&self) -> Result<MinerConfig, String> {
+        if let Some(path) = &self.config_path {
+            let config_file = ConfigFile::from_path(path.as_str())?;
+            let config = Config::from_config_file(config_file)?;
+            Ok(config.miner)
+        } else {
+            Ok(self.miner.clone())
+        }
+    }
```

Comment:
Looking at the invocations of `get_miner_config` (and, for that matter `get_burnchain_config`), it seems like every caller does the same defaulting behavior, so I think it makes sense to just do:

```suggestion
    /// re-read an up-to-date config from this config file's path, or if there is no path,
    /// just return a clone of the current config.
    fn reload_config(&self) -> Result<Config, String> {
        let Some(path) = &self.config_path else {
            return self.clone()
        };
        let config_file = ConfigFile::from_path(path.as_str())?;
        Config::from_config_file(config_file)
    }
    
    /// re-read the up-to-date miner options from the config file's path, or if there is 
    /// no path or it is unreadable, just return the current miner config.
    pub fn get_miner_config(&self) -> MinerConfig {
        let config = self.reload_config().unwrap_or_else(|e| {
             warn!("Failed to reload miner config: {e}");
             self.clone()
        });
        config.miner
    }
```